models/articles.models.js

```diff
@@ -15,21 +15,37 @@ exports.getArticleById = (article_id) => {
     });
 };
 
-exports.getArticles = () => {
-  return db.query(`
-    SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, 
-    COUNT(comments.article_id)::INT 
-    AS comment_count
-    FROM articles
-    LEFT JOIN comments 
-    ON comments.article_id = articles.article_id
-    GROUP BY articles.article_id
-    ORDER BY articles.created_at DESC
-    ;`)
-    .then(({ rows }) => {
-      return rows;
-    });
-};
+exports.getArticles = (topic) => {
+  let queryStart = `  
+  SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, 
+  COUNT(comments.article_id)::INT 
+  AS comment_count
+  FROM articles
+  LEFT JOIN comments 
+  ON comments.article_id = articles.article_id`
+  const queryEnd = `
+  GROUP BY articles.article_id
+  ORDER BY articles.created_at DESC;`
+  let queryTopic = []
+  
+  if(topic){
+  queryStart += `
+  WHERE articles.topic = $1`
+  queryTopic.push(topic)
+  }
+  return db.query(`${queryStart} ${queryEnd}`, queryTopic)
+      .then(({ rows }) => {
+        if(rows.length > 0){
+          return rows;
+        }
+        return Promise.reject({
```

Comment:
This may need to change to pass the 200 test you need to add 👀 
Some topics will be valid but not have any articles yet (like "paper")
This means you'll at some point need to check that the provided topic exists in the database (remember you have an endpoint that gets the topics!)